admin/release.py

```diff
@@ -67,3 +92,212 @@ def make_rpm_version(flocker_version):
         release.extend(remainder)
 
     return rpm_version(version, '.'.join(release))
+
+
+class NotTagged(Exception):
+    """
+    Raised if publishing to production and the published version isn't tagged.
+    """
+
+
+class NotARelease(Exception):
+    """
+    Raised if trying to publish to a version that isn't a release.
+    """
+
+
+class Environments(Names):
+    """
+    The environments that documentation can be published to.
+    """
+    PRODUCTION = NamedConstant()
+    STAGING = NamedConstant()
+
+
+@attributes([
+    'documentation_bucket',
+    'cloudfront_cname',
+    'dev_bucket',
+])
+class DocumentationConfiguration(object):
+    """
+    The configuration for publishing documentation.
+
+    :ivar bytes documentation_bucket: The bucket to publish documentation to.
+    :ivar bytes cloudfront_cname: a CNAME associated to the cloudfront
+        distribution pointing at the documentation bucket.
+    :ivar bytes dev_bucket: The bucket buildbot uploads documentation to.
+    """
+
+DOCUMENTATION_CONFIGURATIONS = {
+    Environments.PRODUCTION:
+        DocumentationConfiguration(
+            documentation_bucket="clusterhq-docs",
+            cloudfront_cname="docs.clusterhq.com",
+            dev_bucket="clusterhq-dev-docs"),
+    Environments.STAGING:
+        DocumentationConfiguration(
+            documentation_bucket="clusterhq-staging-docs",
+            cloudfront_cname="docs.staging.clusterhq.com",
+            dev_bucket="clusterhq-dev-docs"),
+}
+
+
+@do
+def publish_docs(flocker_version, doc_version, environment):
+    """
+    Publish the flocker documentation.
+
+    :param bytes flocker_version: The version of flocker to publish the
+        documentation for.
+    :param bytes doc_version: The version to publish the documentation as.
+    :param Environments environment: The environment to publish the
+        documentation to.
+    :raises NotARelease: Raised if trying to publish to a version that isn't a
```

Comment:
:question: I was initially confused reading this, thinking that it meant "trying to publish a version that...". I can't think of a much better alternative.